File: helpers/cpp/template.cc
#include <chrono>
#include <cmath>
#include <fstream>
#include <iostream>
#include <string>
#include <vector>

using namespace std;

void partTwo(const string &file_path) {
  auto start = chrono::high_resolution_clock::now();
  ifstream file(file_path);
  string input;
  uint64_t answer = 0;

  while (getline(file, input)) {
  }

  auto end = chrono::high_resolution_clock::now();
  auto elapsed = chrono::duration_cast<chrono::milliseconds>(end - start);
  cout << answer << " Time: " << elapsed.count() << " ms" << endl;
}

void partOne(const string &file_path) {
  auto start = chrono::high_resolution_clock::now();
  ifstream file(file_path);
  string input;
  uint64_t answer = 0;

  while (getline(file, input)) {
  }

  auto end = chrono::high_resolution_clock::now();
  auto elapsed = chrono::duration_cast<chrono::milliseconds>(end - start);
  cout << answer << " Time: " << elapsed.count() << " ms" << endl;
}

int main() {
  string test_path = "test.txt";
  string my_path = "input.txt";

  cout << "========PART 1========" << endl;
  cout << "Test: ";
  partOne(test_path);
  // cout << "Input: ";
  // partOne(my_path);

  // cout << endl;

  // cout << "=========PART 2=========" << endl;
  // cout << "Test: ";
  // partTwo(test_path);
  // cout << "Input: ";
  // partTwo(my_path);

  return 0;
};

File: helpers/rust/template.rs
use std::fs;
use std::io::{self, BufRead};
use std::path::Path;
use std::time::Instant;

fn main() {
    let test_path: &str = "src/test.txt";
    let input_path: &str = "src/input.txt";
    println!("========= PART ONE ==========");
    print!("Test ");
    part_one(test_path);
    print!("Input ");
    part_one(input_path);

    println!("========= PART TWO ==========");
    // print!("Test ");
    // part_two(test_path);
    // print!("Input ");
    // part_two(input_path);
}

// Used to read text file line by line

#[allow(dead_code)]
fn part_one(file_path: &str) {
    let now = Instant::now();

    let mut answer: Option<u64> = None;
    // Parse the problem
    if let Ok(lines) = read_lines(file_path) {
        for line in lines.map_while(Result::ok) {}
    }

    let elapsed = now.elapsed();
    println!("Answer: {} Time: {} ms", answer, elapsed.as_millis());
}

#[allow(dead_code)]
fn part_two(file_path: &str) {
    let now = Instant::now();

    let mut answer: Option<u64> = None;
    // Parse the problem
    if let Ok(lines) = read_lines(file_path) {
        for line in lines.map_while(Result::ok) {}
    }

    let elapsed = now.elapsed();
    println!("Answer: {} Time: {} ms", answer, elapsed.as_millis());
}
